src/main/java/adris/altoclef/tasks/container/LootContainerTask.java

```diff
@@ -57,34 +61,49 @@ protected Task onTick(AltoClef mod) {
         }
         Optional<Slot> optimal = getAMatchingSlot(mod);
         if (optimal.isEmpty()) {
+            _weDoneHere = true;
             return null;
         }
-        setDebugState("Looting a container for all of their " + target.toString());
+        setDebugState("Looting items: " + targets);
         return new ClickSlotTask(optimal.get());
     }
 
     @Override
     protected void onStop(AltoClef mod, Task task) {
-        if (!_wasProtected) {
-            mod.getBehaviour().removeProtectedItems(target);
+        if (ContainerType.screenHandlerMatches(ContainerType.CHEST)) StorageHelper.closeScreen();
+        for (Item item : _protected) {
+            mod.getBehaviour().removeProtectedItems(item);
         }
         mod.getBehaviour().pop();
     }
 
     @Override
     protected boolean isEqual(Task other) {
-        return other instanceof LootContainerTask && target == ((LootContainerTask) other).target;
+        return other instanceof LootContainerTask && targets == ((LootContainerTask) other).targets;
     }
 
     private Optional<Slot> getAMatchingSlot(AltoClef mod) {
-        List<Slot> slots = mod.getItemStorage().getSlotsWithItemContainer(target);
-        if (slots.isEmpty()) return Optional.empty();
-        else return Optional.ofNullable(slots.get(0));
+        for (Item item : targets) {
+            List<Slot> slots = mod.getItemStorage().getSlotsWithItemContainer(item);
+            if (!slots.isEmpty()) for (Slot slot : slots) {
+                boolean hasBinding = false;
+                for (NbtElement elm : StorageHelper.getItemStackInSlot(slot).getEnchantments()) {
+                    NbtCompound comp = (NbtCompound) elm;
+                    if (comp.getString("id").equals("minecraft:binding_curse")) {
+                        hasBinding = true;
+                        break;
+                    }
+                }
+                if (!hasBinding) return Optional.of(slot);
+            }
+        }
+        return Optional.empty();
     }
 
     @Override
     public boolean isFinished(AltoClef mod) {
-        return getAMatchingSlot(mod).isEmpty();
+        return _weDoneHere || (ContainerType.screenHandlerMatches(ContainerType.CHEST) &&
```

Comment:
What if we're grabbing from a furnace/barrel? There should be a `ContainerType.getOpenScreenType` function or something in `StorageHelper`

Reply:
I dunno I only intended this for chests

Reply:
Basically I wanted this code to make sure that we are in a container, how would I check that?

Reply:
Also you don't even have anything for barrel lol